### PR TITLE
[TS-676] Add interface for configuring a client before connecting to …

### PR DESCRIPTION
**Pull request by OptimumCode**
…the system.

* add a parameter with default header to DefaultHandlerSettings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Service will also automatically connect prior to message send if it wasn't conne
 Event handler can be configured by changing following properties in the `handlerSettings` block of the main configuration:
 
 + **pingInterval** - interval for sending ping-messages in ms (`30000` by default)
++ **defaultHeaders** - the list of headers and their values to add to the HTTP handshake request (the list is **empty** by default)
```
Comment (cordwelt) [inline README.md]:
```suggestion
+ **defaultHeaders** - map of headers and their values to add to the HTTP handshake request (the map is **empty** by default)
```

File: README.md [hunk 2]
```diff
@@ -94,6 +101,11 @@ spec:
 
 ## Changelog
 
+### v0.2.0
```
Comment (cordwelt) [inline README.md:104]:
```suggestion
### v0.2.0

#### Added
```

File: src/main/kotlin/com/exactpro/th2/ws/client/api/IClientSettings.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Provides the ability to configure the [IClient] implementation
+ */
+interface IClientSettings {
+    /**
+     * Adds HTTP header to the WebSocket client handshake request
+     */
+    fun addHeader(name: String, value: String)
+}
+
+fun IClientSettings.addHeaders(name: String, values: Collection<String>) {
```
Comment (cordwelt) [inline src/main/kotlin/com/exactpro/th2/ws/client/api/IClientSettings.kt]: Maybe be move it to the interface?

File: src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt [hunk 4]
```diff
@@ -34,6 +36,10 @@ class DefaultHandler : IHandler {
         }
     }
 
+    override fun preOpen(clientSettings: IClientSettings) {
+        settings.defaultHeaders.forEach { (header, values) -> clientSettings.addHeaders(header, values) }
```
Comment (cordwelt) [inline src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt]:
```suggestion
        settings.defaultHeaders.forEach(clientSettings::addHeaders)
```

**Review by cordwelt — CHANGES_REQUESTED** [review]
Please also update versions [here](https://github.com/th2-net/th2-conn-ws-client/blob/ts-676/README.md#websocket-client-v011) and [here](https://github.com/th2-net/th2-conn-ws-client/blob/ts-676/README.md#deployment-via-infra-mgr)

File: src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt (hunk 4 above)
Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt]: This one will work only if we move the `addHeaders` method to the interface. If we leave it as extension method the function signature won't match the required one

File: src/main/kotlin/com/exactpro/th2/ws/client/api/IClientSettings.kt (hunk 3 above)
Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/ws/client/api/IClientSettings.kt]: I tried to keep the interface as simple as possible. But yes, we can move it to the interface itself as a default method

File: src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt (hunk 4 above)
Comment (cordwelt) [reply src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt]: I've tried it in IDE (didn't run the code though) and it has no problems in resolving a proper method

**Review by cordwelt — APPROVED** [review]
Please use squash when merging this PR

File: src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt (hunk 4 above)
Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt]: Hm...yes, you are right. I have tried it again and everything compiles. Probably, I did something wrong the first time

**Pull request merged**